[Docs] Guide on setting query context (#18179)
apache/druid: docs/querying/query-context-reference.md

  ~ under the License.
  -->

The query context provides runtime configuration for individual queries in Apache Druid. Each parameter in the query context controls a specific aspect of query behavior—from execution timeouts and resource limits to caching policies and processing strategies.

> Note that setting query context will override both the default value and the runtime properties value in the format of

Reviewer: ```suggestion
Query context overrides both the default value and the runtime properties value in the format of
```
I would also suggest that it is not a "note'. Also avoid future.

`druid.query.default.context.{property_key}` (if set). 

This reference contains context parameters organized by their scope: 

- **General parameters**: Applies to all query types.
- **Parameters by query type**: Applies to the specific type of query, such as TopN, Timeseries, or GroupBy.
- **Vectorization parameters**: Controls vectorized query execution for supported query types.

> To learn how to set query context, see [Set query context](./set-query-context.md).

Reviewer 2: "how to set **the** query context"

  ~ under the License.
  -->

> Query context provides runtime configuration for individual queries in Apache Druid. You can configure query context through context parameters. Each parameter in the query context controls a specific aspect of query behavior—from execution timeouts and resource limits to caching policies and processing strategies.

Reviewer 2: ```suggestion
The query context provides runtime configuration for individual queries in Apache Druid. Each parameter in the query context controls a specific aspect of query behavior—from execution timeouts and resource limits to caching policies and processing strategies.
```

> Query context parameters can be specified in the following ways:

Reviewer 2: Lines 28-32 can be removed since it's now covered in the new doc

Note that setting query context will override both the default value and the runtime properties value in the format of
`druid.query.default.context.{property_key}` (if set). 

> This reference contains all available context parameters organized by their scope: 

Reviewer 2: ```suggestion
This reference contains context parameters organized by their scope: 
```

- **General parameters**: Apply to all query types and control execution behavior, resource limits, and caching
- **Query type-specific parameters**: Available only for certain query types like TopN, Timeseries, or GroupBy
- **Vectorization parameters**: Control vectorized query execution for supported query types

> 

Reviewer 2: Also link to the MSQ context parameters table: https://druid.apache.org/docs/latest/multi-stage-query/reference#context-parameters

> For instructions on how the different approaches for set query context in your queries, see [Set the Query Context](./set-query-context.md).

Reviewer 2: ```suggestion
To learn how to set the query context, see [Set query context](./set-query-context.md).
```

Reviewer 2: Suggest re-reviewing all references to the new doc to ensure
1. correct title
2. correct capitalization
